### PR TITLE
Update Django REST Framework to version 3.0

### PR DESCRIPTION
**Pull request by noefroidevaux**

### CODE REVIEW

File: api/serializers.py [hunk 1]
```diff
@@ -4,46 +4,63 @@
...
+    def update(self, instance, validated_data):
+        pass
 
     class Meta:
         model = get_user_model()
         fields = ('id', 'username', 'first_name', 'last_name')
+
+
+class GameSerializer(serializers.Serializer):
+    winner_id = serializers.IntegerField(required=True)
```
Comment (sephii) [inline api/serializers.py]: Use `PrimaryKeyRelatedField`s for these (http://www.django-rest-framework.org/api-guide/relations/)

File: api/serializers.py [hunk 2]
```diff
@@ -4,46 +4,63 @@
...
 
     class Meta:
-        model = Score
-        fields = ('id', 'team', 'competition', 'score', 'stdev')
+        model = Team
+        fields = ('id', 'name', 'users', 'competitions')
+
 
+class UserSerializer(serializers.ModelSerializer):
+    def create(self, validated_data):
```
Comment (sephii) [inline api/serializers.py]: Is there any reason these are empty? Is it to disallow user creation/update? Could you put a comment for that?

File: api/serializers.py [hunk 3]
```diff
@@ -4,46 +4,63 @@
...
+        }
+
+    def create(self, validated_data):
+        winner = Team.objects.get(pk=validated_data['winner_id'])
+        loser = Team.objects.get(pk=validated_data['loser_id'])
+        competition = Competition.objects.get(pk=validated_data['competition_id'])
+
+        return Game.objects.announce(winner.users.all()[0], loser.users.all()[0], competition)
+
+    def update(self, instance, validated_data):
```
Comment (sephii) [inline api/serializers.py]: Same comment as on `UserSerializer`

File: api/urls.py [hunk 4]
```diff
@@ -1,42 +1,22 @@
...
-                            base_name='competitions-games', parents_query_lookups=['competitions'])
-competition_router.register(r'scores', ScorePerCompetitionViewSet,
-                            base_name='competitions-scores', parents_query_lookups=['competition'])
-
-user_router = router.register(r'users', UserViewSet)
-user_router.register(r'competitions', CompetitionPerUserViewSet,
-                     base_name='users-competitions', parents_query_lookups=['users'])
+from django.conf.urls import url, include
+from rest_framework import routers
+from api.views import CompetitionViewSet, TeamViewSet, UserViewSet, GameViewSet, ScoreViewSet
```
Comment (sephii) [inline api/urls.py]: Use relative import (ie. `from .views`)

File: game/models.py [hunk 5]
```diff
@@ -565,9 +568,15 @@ class Meta:
 
     def __unicode__(self):
         return '[%s] %s: mu = %s, s = %s' % (self.competition.name,
-                                             self.team.get_name(), self.score,
+                                             self.get_team_name, self.score,
```
Comment (sephii) [inline game/models.py]: Should be `get_team_name()`

File: api/serializers.py (hunk 2 above)
Comment (noefroidevaux) [inline api/serializers.py]: In fact, there is no reason because we use viewsets.ReadOnlyModelViewSet in views. I'll remove the methods.

**Pull request merged**